pkts-core/src/main/java/io/pkts/packet/impl/IPv6PacketImpl.java

```diff
@@ -0,0 +1,265 @@
+/**
+ *
+ */
+package io.pkts.packet.impl;
+
+import io.pkts.buffer.Buffer;
+import io.pkts.buffer.Buffers;
+import io.pkts.framer.FramingException;
+import io.pkts.framer.TCPFramer;
+import io.pkts.framer.UDPFramer;
+import io.pkts.packet.IPv6Packet;
+import io.pkts.packet.Packet;
+import io.pkts.packet.PacketParseException;
+import io.pkts.protocol.Protocol;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/**
+ * @author epall@google.com
+ */
+public final class IPv6PacketImpl extends AbstractPacket implements IPv6Packet {
+    public static final int FIXED_HEADER_LENGTH = 40;
+
+    private static final UDPFramer udpFramer = new UDPFramer();
+
+    private static final TCPFramer tcpFramer = new TCPFramer();
+
+    private final Buffer headers;
+
+    private final int nextProtocol;
+
+    public IPv6PacketImpl(final Packet parent, final Buffer headers, int nextProtocol, final Buffer payload) {
+        super(Protocol.IPv6, parent, payload);
+        assert parent != null;
+        assert headers != null;
+        this.headers = headers;
+        this.nextProtocol = nextProtocol;
+    }
+
+    /**
+     * Get the raw source ip.
+     *
+     * Note, these are the raw bits and should be treated as such. If you really
+     * want to print it, then you should treat it as unsigned
+     *
+     * @return
+     */
+    @Override
+    public byte[] getRawSourceIP() {
+        final Buffer tmp = Buffers.createBuffer(128 / 8);
+        this.headers.getBytes(8, tmp);
+        return tmp.getArray();
+    }
+
+    @Override
+    public void setSourceIP(String sourceIp) {
+        // TODO
+        throw new RuntimeException("Not implemented");
+    }
+
+    /**
+     *
+     * {@inheritDoc}
+     */
+    @Override
+    public String getSourceIP() {
+        try {
+            return InetAddress.getByAddress(getRawSourceIP()).getHostAddress();
+        } catch (UnknownHostException e) {
+            return null;
```

Comment:
returning null seems kind of unexpected. I guess one of the issues is that the InetAddress will perform a DNS query (potentially) and would be nice to avoid that.

Reply:
It'll perform a DNS query? I think because we're feeding it an address, it'll just do the calculation offline and return a String. The situation it might throw `UnknownHostException` is when the source IP is malformed, or some other crazy edge case. Should we throw a `RuntimeException` instead of returning null?